locator/token_metadata.hh

```diff
@@ -309,6 +309,10 @@ public:
 
     const std::unordered_set<NodeId>& get_all_endpoints() const;
 
+    template <typename T = NodeId>
+    requires std::is_same_v<T, locator::host_id>
```

Comment:
Making this a template doesn't do anything, IIUC.

Did you want to constraint the outer `NodeId` parameter this way? If so, this doesn't achieve the goal. `typename T = NodeId` is only a default value for `T`. You can still call `get_all_ips<locator::host_id>`  on `generic_token_metadata<gms::inet_address>`. Compilation will fail when trying to compile implementation of `get_all_ips` (in the same way it would fail if you didn't make this a template; only the compilation error will be less readable)